--- openpetra.js.before/csharp/ICT/Common/IO/HTTPUtils.cs	2014-08-30 15:06:36.000000000 +0200
+++ openpetra.js/csharp/ICT/Common/IO/HTTPUtils.cs	2014-08-30 15:03:00.000000000 +0200
@@ -95,7 +95,9 @@
             protected override WebRequest GetWebRequest(Uri address)
             {
                 TLogging.LogAtLevel(1, "GetWebRequest: got called for URI: " + address.ToString());
-                WebRequest request = base.GetWebRequest(address);
+                HttpWebRequest request = (HttpWebRequest)base.GetWebRequest(address);
+
+                request.UserAgent="Mozilla/4.0 (compatible; MSIE 7.0; Windows NT 6.1; WOW64; Trident/6.0; OpenPetraFatClient)";
 
                 request.Timeout = Convert.ToInt32(
                     TimeSpan.FromMinutes(TAppSettingsManager.GetInt32("WebRequestTimeOutInMinutes", 15)).
