--- openpetra.js.before/inc/template/src/ClientServerGlue/ServerGlue.cs	2014-08-30 15:06:37.000000000 +0200
+++ openpetra.js/inc/template/src/ClientServerGlue/ServerGlue.cs	2014-08-30 15:01:03.000000000 +0200
@@ -38,6 +38,11 @@
 {
     private static SortedList<string, object> FUIConnectors = new SortedList<string, object>();
 
+    private bool isJSClient()
+    {
+       return HttpContext.Current!=null && !HttpContext.Current.Request.UserAgent.Contains("OpenPetra");
+    }
+
     /// <summary>
     /// constructor, which is called for each http request
     /// </summary>
