--- openpetra.js.before/csharp/ICT/Common/Remoting/Server/Session.cs	2014-08-29 15:39:26.000000000 +0200
+++ openpetra.js/csharp/ICT/Common/Remoting/Server/Session.cs	2014-08-29 15:37:58.000000000 +0200
@@ -52,6 +52,12 @@
         {
             string sessionID = GetSessionID();
 
+            if ((sessionID != string.Empty) && !FSessionObjects.ContainsKey(sessionID))
+            {
+                HttpContext.Current.Request.Cookies.Remove("OpenPetraSessionID");
+                sessionID = GetSessionID();
+            }
+
             if (sessionID == string.Empty)
             {
                 sessionID = Guid.NewGuid().ToString();
