--- openpetra.js.before/csharp/ICT/Common/Logging.cs	2014-08-29 15:39:26.000000000 +0200
+++ openpetra.js/csharp/ICT/Common/Logging.cs	2014-08-29 15:37:58.000000000 +0200
@@ -277,11 +277,11 @@
                 // only in Debugmode write the messages for the statusbar also on the console (e.g. reporting progress)
                 || (((ALoggingType & TLoggingType.ToStatusBar) != 0) && (TLogging.DebugLevel == TLogging.DEBUGLEVEL_TRACE)))
             {
-                Console.Error.WriteLine(Utilities.CurrentTime() + "  " + Text);
+                Console.WriteLine(Utilities.CurrentTime() + "  " + Text);
 
                 if ((TLogging.Context != null) && (TLogging.Context.Length != 0))
                 {
-                    Console.Error.WriteLine("  Context: " + TLogging.Context);
+                    Console.WriteLine("  Context: " + TLogging.Context);
                 }
             }
 
