--- openpetra.js.before/webserver/Samples/DemosForOpenPetraInBrowser/JQueryDesktopDemo/Default.aspx	2014-08-29 15:39:26.000000000 +0200
+++ openpetra.js/webserver/Samples/DemosForOpenPetraInBrowser/JQueryDesktopDemo/Default.aspx	2014-08-29 15:37:59.000000000 +0200
@@ -1,4 +1,6 @@
-﻿<!DOCTYPE HTML>
+﻿<%@ Page Language="C#" %>
+
+<!DOCTYPE HTML>
 <html>
 <head>
 <meta http-equiv="Content-Type" content="text/html; charset=utf-8">
@@ -70,4 +72,4 @@
 <script type="text/javascript" src="js/json2.js"></script> <!-- TODO minify https://raw.github.com/douglascrockford/JSON-js/master/json2.js -->
 <script type="text/javascript" src="js/login.js"></script>
 </body>
-</html>
\ No newline at end of file
+</html>
